### ISSUE TITLE
Print detailed memory reclamation diagnostics on stalls

### ISSUE THREAD

**Issue by tgrabiec** [scylladb/scylla#4186]
Reactor stalls often occur during memory reclamation. There can be many causes of that:
  - there's a large allocation request and lots of data needs to be evicted
  - the LSA reserves are unusually high
  - LSA regions are in a state such that reclamation is not efficient, e.g. segments have large occupancy and a lot of items need to be reclaimed before segment compaction can start
  - Eviction of objects is slow due to characteristics of the data structures involved.

In order to make pinpointing the problem easier, the memory reclamation should log diagnostics when it detects unusually long reclamation, including:
  - the amount of memory or LSA segments requested to be reclaimed
  - occupancy of LSA regions involved, before and after
  - the number of LSA segments compacted, migrated and allocated
  - the number (and kinds) of objects migrated and evicted

Detection can be tight to the preemption counter, like in the case of the stall detector.

**Comment by bhalevy** [comment]
@espindola please put this on your queue after/along with #6365

**Comment by slivne** [comment]
Whats left to close this ?

**Comment by bhalevy** [comment]
> Whats left to close this ?

@tgrabiec / @cmm ?

**Comment by bhalevy** [comment]
ping @tgrabiec / @cmm
is there anything left to be done here or can the issue be closed?

**Comment by cmm** [comment]
I don't believe the "the number (and kinds) of objects migrated and evicted" part is covered, for example

**Comment by slivne** [comment]
@bhalevy / @cmm ping is this done ? can we close ?
